editor/src/components/canvas/dom-lookup.ts

```diff
@@ -34,17 +35,20 @@ export function findFirstParentWithValidUID(
   validTemplatePathsForLookup: Array<string> | 'no-filter',
   target: Element,
 ): string | null {
-  const uid = getDOMAttribute(target, 'data-uid')
-  const originalUid = getDOMAttribute(target, 'data-utopia-original-uid')
+  const uids = getUIDsOnDomELement(target)
+  const originalUid = getDOMAttribute(target, 'data-utopia-original-uid') // TODO BALAZS What's up with original-uid?
```

Comment:
Heads up, I'm binning this off right now and will open a separate PR for that to unblock this PR.